stored_requests/backends/http_fetcher/fetcher.go

```diff
@@ -163,7 +167,39 @@ func (fetcher *HttpFetcher) FetchAccount(ctx context.Context, accountID string)
 			DataType: "Account",
 		}}
 	}
-	return accountJSON, nil
+	// accountID resolved to a valid account, merge with AccountDefaults for a complete config
+	account := &config.Account{}
+	completeJSON, err := jsonpatch.MergePatch(accountDefaultsJSON, accountJSON)
```

Comment:
With leaving the return type as raw JSON for `FetchAccount`, you can just return after this step and move the rest of the GDPR related backwards compatibility logic in the original `GetAccount` function itself. This will minimize the duplicate code needed in each of the fetcher implementation.

Reply:
Done